src/builtins/amp-layout/amp-layout.js

```diff
@@ -34,18 +35,29 @@ class AmpLayout extends BaseElement {
 
   /** @override */
   buildCallback() {
-    if (this.getLayout() == Layout.CONTAINER) {
-      return;
-    }
-    const container = this.win.document.createElement('div');
-    applyFillContent(container);
-    realChildNodes(this.element).forEach((child) => {
-      container.appendChild(child);
-    });
-    this.element.appendChild(container);
+    buildDom(this.win.document, this.element);
   }
 }
 
+/**
+ *
+ * @param {!Document} document
+ * @param {!Element} element
+ */
+export function buildDom(document, element) {
+  const {layout} = getEffectiveLayout(element);
+  if (layout == Layout.CONTAINER) {
+    return;
+  }
+
+  const container = document.createElement('div');
+  applyFillContent(container);
+  realChildNodes(element).forEach((child) => {
```

Comment:
Could you use the `copyChildren` helper from that other PR here?

Reply:
We would need a `moveChildren`, or `reparent` instead of copy